friendshipper/src-tauri/src/repo/operations/reset.rs

```diff
@@ -1,12 +1,177 @@
-use crate::engine::EngineProvider;
-use crate::state::AppState;
-use axum::extract::State;
-use ethos_core::types::errors::CoreError;
-
-pub async fn reset_repo<T>(State(state): State<AppState<T>>) -> Result<(), CoreError>
-where
-    T: EngineProvider,
-{
-    let branch = state.repo_config.read().trunk_branch.clone();
-    state.git().hard_reset(&branch).await.map_err(|e| e.into())
-}
+use anyhow::{anyhow, Context};
+use async_trait::async_trait;
+use axum::extract::{Path, State};
+
+use ethos_core::clients::aws::ensure_aws_client;
+use ethos_core::clients::git;
+use ethos_core::types::config::{RepoConfig, UProject};
+use ethos_core::types::errors::CoreError;
+use ethos_core::worker::{Task, TaskSequence};
+
+use crate::engine::EngineProvider;
+use crate::repo::operations::{DownloadDllsOp, StatusOp, UpdateEngineOp};
+use crate::state::AppState;
+use ethos_core::storage::config::Project;
+use ethos_core::storage::ArtifactStorage;
+use ethos_core::AWSClient;
+use std::path::PathBuf;
+use tokio::sync::oneshot::error::RecvError;
+
+pub async fn reset_repo<T>(State(state): State<AppState<T>>) -> Result<(), CoreError>
+where
+    T: EngineProvider,
+{
+    let branch = state.repo_config.read().trunk_branch.clone();
+    state.git().hard_reset(&branch).await.map_err(|e| e.into())
+}
+
+#[derive(Clone)]
+pub struct ResetToCommitOp {
+    pub commit: String,
+    pub git_client: git::Git,
+}
+
+#[async_trait]
+impl Task for ResetToCommitOp {
+    async fn execute(&self) -> Result<(), CoreError> {
+        self.git_client
+            .run(&["reset", "--hard", &self.commit], Default::default())
```

Comment:
`--hard` will get rid of all local changes. is that the intent here?

Reply:
using `--keep` 👍